[2025-09-16T08:42:16.399+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: opensky_data_pipeline_dag.run_bronze_load manual__2025-09-16T08:41:47.125219+00:00 [queued]>
[2025-09-16T08:42:16.401+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: opensky_data_pipeline_dag.run_bronze_load manual__2025-09-16T08:41:47.125219+00:00 [queued]>
[2025-09-16T08:42:16.401+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 1
[2025-09-16T08:42:16.404+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): run_bronze_load> on 2025-09-16 08:41:47.125219+00:00
[2025-09-16T08:42:16.407+0000] {standard_task_runner.py:60} INFO - Started process 1491 to run task
[2025-09-16T08:42:16.409+0000] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'opensky_data_pipeline_dag', 'run_bronze_load', 'manual__2025-09-16T08:41:47.125219+00:00', '--job-id', '14', '--raw', '--subdir', 'DAGS_FOLDER/opensky_data_pipeline_dag.py', '--cfg-path', '/tmp/tmpe5aya398']
[2025-09-16T08:42:16.410+0000] {standard_task_runner.py:88} INFO - Job 14: Subtask run_bronze_load
[2025-09-16T08:42:16.429+0000] {task_command.py:423} INFO - Running <TaskInstance: opensky_data_pipeline_dag.run_bronze_load manual__2025-09-16T08:41:47.125219+00:00 [running]> on host 64220da1adfe
[2025-09-16T08:42:16.529+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='opensky_data_pipeline_dag' AIRFLOW_CTX_TASK_ID='run_bronze_load' AIRFLOW_CTX_EXECUTION_DATE='2025-09-16T08:41:47.125219+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-09-16T08:41:47.125219+00:00'
[2025-09-16T08:42:16.574+0000] {bronze_load.py:27} INFO - Loaded 487952 rows from Parquet: /opt/airflow/scripts/opensky/../../data/kafka_parquet/opensky_batch_20250916_084215.parquet
[2025-09-16T08:42:16.574+0000] {bronze_load.py:28} INFO - Beginning to extract and load data into bronze_info table.
[2025-09-16T08:42:16.581+0000] {base.py:83} INFO - Using connection ID 'postgres_default' for task execution.
[2025-09-16T08:42:16.661+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/etl/bronze/opensky/bronze_load.py", line 40, in insert_into_bronze_ddl
    cur.execute(insert_query, full_row)
psycopg2.errors.InvalidTextRepresentation: invalid input syntax for type json
LINE 24: ...aN'::float, NULL, 'NaN'::float, '4362', false, 0, '["a53eef"...
                                                              ^
DETAIL:  Token "NaN" is invalid.
CONTEXT:  JSON data, line 1: ... 1758011337.0, 1758011337, -78.7367, 42.9373, NaN...

[2025-09-16T08:42:16.665+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=opensky_data_pipeline_dag, task_id=run_bronze_load, execution_date=20250916T084147, start_date=20250916T084216, end_date=20250916T084216
[2025-09-16T08:42:16.668+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 14 for task run_bronze_load (invalid input syntax for type json
LINE 24: ...aN'::float, NULL, 'NaN'::float, '4362', false, 0, '["a53eef"...
                                                              ^
DETAIL:  Token "NaN" is invalid.
CONTEXT:  JSON data, line 1: ... 1758011337.0, 1758011337, -78.7367, 42.9373, NaN...
; 1491)
[2025-09-16T08:42:16.682+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-09-16T08:42:16.691+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
